### ISSUE TITLE
Italics in subtitles

### ISSUE THREAD

**Issue by roberttucker** [ableplayer/ableplayer#193]
I have valid .vtt track files which work except that italicized text (\<i\>\</i\> tags in the .vtt flie) is not shown in the subtitles.

Is that a bug or known issue or have I got something wrong somewhere?

**Comment by terrill** [comment]
That's a bug in our WebVTT parser. I think <i> may be a relatively recent addition to the WebVTT spec and we haven't put any effort to staying current with changes in the spec. This particular issue should be easy enough to fix, but we're likely to fall behind in other areas as well unless someone wants to step up and claim responsibility for keeping our parser synced with the evolving WebVTT spec. Otherwise we'll probably wait until it's finalized as an official W3C recommendation.

**Comment by roberttucker** [comment]
I don't understand the full process but obviously it's parsing the tags out as they don't appear in the subtitles. Italics are very often used for off-screen narration and sometimes people like to use different colours for different speakers. If just the class tag were made available, would that not facilitate both these (and bold and underline)?

Would it be possible to integrate C Giffard's Captionator parser in some way? (Again not fully realizing all that might be involved.)

**Comment by terrill** [comment]
@roberttucker I suspect we can learn some things from [C Giffard's Captionator parser](https://github.com/cgiffard/Captionator) - thanks for the heads-up about that. However, I suspect he may ultimately face the same issue we do. His latest updates were in 2013, whereas the WebVTT spec is constantly evolving (lots of updates throughout 2015 and more already in 2016). 

- [Current WebVTT spec](https://w3c.github.io/webvtt/)
- [WebVTT Commits on GitHub](https://github.com/w3c/webvtt/commits)

Anyway, it won't be difficult to add support for <i>, <b>, and <u> in Able Player.  That's on my todo list now.

**Comment by terrill** [comment]
Also, we do have full support for WebVTT on our roadmap for future development:  #13

**Comment by roberttucker** [comment]
I've sort of got italics to display by adding:

```
else if (component.type === 'i')
...
...
```

similar to the ones you have for 'v' in caption.js and metadata.js, recompiling and adding:

```
else if (component.type === 'i') {        
  result.push(component.value.italics());
...
...
```

in the two places you have

```
else if (component.type === 'v') {
   result.push('[' + component.value + ']');
```

in ableplayer.js.

It seems to work on a test – the only thing is that unusual tags are needed:
`<i {italic text string}>`
in the .vtt!

**Comment by roberttucker** [comment]
Oh, and deleted the 'i', from:

if ($.inArray(token.tagName, ['c', ~~'i',~~ 'b', 'u', 'ruby']) !== -1) {

in ableplayer.js.

[Sorry, for the kludge but thought I might let you know there's not too much hurry for an italics fix here.]

**Comment by terrill** [comment]
@roberttucker this isn't valid WebVTT: 
`<i {italic text string}>` 

This would be valid: 
`<i>{italic text string}</i>`

That said, if your solution works for your purposes, no problem. For the final Able Player solution though we'll want to be sure to honor the spec. I'm closing this issue now as it will ultimately be addressed in #13
